From 8b493063ad15668551214596e4e2ee7f5548989b Mon Sep 17 00:00:00 2001
From: Alvaro Soliverez <asoliverez@kde.org>
Date: Sun, 24 Nov 2013 13:55:11 -0300
Subject: [PATCH] If using ISO for week numbering, first day of week must be
 Monday Use KLocale for week numbers instead of QDate's week methods

Specific patch for 3rd party libkdgantt
---
 3rdparty/kdgantt/CMakeLists.txt          |  5 +++--
 3rdparty/kdgantt/kdganttdatetimegrid.cpp | 19 +++++++++++++++++--
 2 files changed, 20 insertions(+), 4 deletions(-)

diff --git a/3rdparty/kdgantt/CMakeLists.txt b/3rdparty/kdgantt/CMakeLists.txt
index 2108158..f6afcba 100644
--- a/3rdparty/kdgantt/CMakeLists.txt
+++ b/3rdparty/kdgantt/CMakeLists.txt
@@ -1,5 +1,6 @@
 include_directories(
     ${QT_INCLUDES}
+    ${KDE4_INCLUDES}
     ${KDGANTT_INCLUDES}
 )
 
@@ -31,8 +32,8 @@ set( kdgantt_LIB_SRCS
 
 kde4_add_library(calligrakdgantt SHARED ${kdgantt_LIB_SRCS})
 
-target_link_libraries(calligrakdgantt ${QT_QTGUI_LIBRARY} ${QT_QTCORE_LIBRARY})
-target_link_libraries(calligrakdgantt LINK_INTERFACE_LIBRARIES ${QT_QTGUI_LIBRARY} ${QT_QTCORE_LIBRARY})
+target_link_libraries(calligrakdgantt ${KDE4_KDECORE_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QTCORE_LIBRARY})
+target_link_libraries(calligrakdgantt LINK_INTERFACE_LIBRARIES ${KDE4_KDECORE_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QTCORE_LIBRARY})
 
 set_target_properties(calligrakdgantt PROPERTIES
     VERSION ${GENERIC_CALLIGRA_LIB_VERSION}
diff --git a/3rdparty/kdgantt/kdganttdatetimegrid.cpp b/3rdparty/kdgantt/kdganttdatetimegrid.cpp
index 79c405c..9ee56aa 100644
--- a/3rdparty/kdgantt/kdganttdatetimegrid.cpp
+++ b/3rdparty/kdgantt/kdganttdatetimegrid.cpp
@@ -25,6 +25,9 @@
 
 #include "kdganttabstractrowcontroller.h"
 
+#include <kcalendarsystem.h>
+#include <kdeversion.h>
+
 #include <QApplication>
 #include <QDateTime>
 #include <QPainter>
@@ -700,6 +703,8 @@ void DateTimeGrid::paintHourScaleHeader( QPainter* painter,  const QRectF& heade
 void DateTimeGrid::paintDayScaleHeader( QPainter* painter,  const QRectF& headerRect, const QRectF& exposedRect,
                                 qreal offset, QWidget* widget )
 {
+    const KCalendarSystem * calendar = KGlobal::locale()->calendar();
+
     // For starters, support only the regular tab-per-day look
     QStyle* style = widget?widget->style():QApplication::style();
 
@@ -732,7 +737,11 @@ void DateTimeGrid::paintDayScaleHeader( QPainter* painter,  const QRectF& header
         QStyleOptionHeader opt;
         opt.init( widget );
         opt.rect = QRectF( x2-offset, headerRect.top(), dayWidth()*7., headerRect.height()/2. ).toRect();
-        opt.text = QString::number( dt.date().weekNumber() );
+#if KDE_IS_VERSION(4,7,0)
+        opt.text = QString::number( calendar->week(dt.date()) );
+#else
+        opt.text = QString::number( calendar->weekNumber(dt.date()) );
+#endif
         opt.textAlignment = Qt::AlignCenter;
         // NOTE:CE_Header does not honor clipRegion(), so we do the CE_Header logic here
         style->drawControl( QStyle::CE_HeaderSection, &opt, painter, 0 ); //NOTE: using widget will loose background when printing
@@ -750,6 +759,8 @@ void DateTimeGrid::paintDayScaleHeader( QPainter* painter,  const QRectF& header
 void DateTimeGrid::paintWeekScaleHeader( QPainter* painter,  const QRectF& headerRect, const QRectF& exposedRect,
                                         qreal offset, QWidget* widget )
 {
+    const KCalendarSystem * calendar = KGlobal::locale()->calendar();
+
     QStyle* style = widget?widget->style():QApplication::style();
 
     // Paint a section for each week
@@ -763,7 +774,11 @@ void DateTimeGrid::paintWeekScaleHeader( QPainter* painter,  const QRectF& heade
         QStyleOptionHeader opt;
         opt.init( widget );
         opt.rect = QRectF( x-offset, headerRect.top()+headerRect.height()/2., dayWidth()*7, headerRect.height()/2. ).toRect();
-        opt.text = QString::number( dt.date().weekNumber() );
+#if KDE_IS_VERSION(4,7,0)
+        opt.text = QString::number( calendar->week(dt.date()) );
+#else
+        opt.text = QString::number( calendar->weekNumber(dt.date()) );
+#endif
         opt.textAlignment = Qt::AlignCenter;
         // NOTE:CE_Header does not honor clipRegion(), so we do the CE_Header logic here
         style->drawControl( QStyle::CE_HeaderSection, &opt, painter, 0 ); //NOTE: using widget will loose background when printing
-- 
1.8.4

